README
README file for Module::Metadata::Changes.

See also: Changes.txt.

Warning: WinZip 8.1 and 9.0 both contain an 'accidental' bug which stops
them recognizing POSIX-style directory structures in valid tar files.
You are better off using a reliable tool such as InfoZip:
ftp://ftp.info-zip.org/pub/infozip/

1 Installing from a Unix-like distro
------------------------------------
shell>gunzip Module-Metadata-Changes-1.00.tgz
shell>tar mxvf Module-Metadata-Changes-1.00.tar

On Unix-like systems, assuming you have installed Module::Build V 0.25+:

shell>perl Build.PL
shell>./Build
shell>./Build test
shell>./Build install

On MS Windows-like systems, assuming you have installed Module::Build V 0.25+:

shell>perl Build.PL
shell>perl Build
shell>perl Build test
shell>perl Build install

Alternately, without Module::Build, you do this:

Note: 'make' on MS Windows-like systems may be called 'nmake' or 'dmake'.

shell>perl Makefile.PL
shell>make
shell>make test
shell>su              (for Unix-like systems)
shell>make install
shell>exit            (for Unix-like systems)

On all systems:

Run Changes.pm through you favourite pod2html translator.

If you are using my fancy-pom2.pl, with its 'default.css' file installed in
/apache2/htdocs/assets/css/, you'd do:

shell>perl fancy-pom2.pl html -css Changes.pm > /apache2/htdocs/assets/Changes.html

or perhaps something like:

shell>perl fancy-pom2.pl html -css Changes.pm > /perl/html/site/lib/Module/Metadata/Changes.html

2 Installing from an ActiveState distro
---------------------------------------
shell>unzip Module-Metadata-Changes-1.00.zip
shell>ppm install --location=. Module-Metadata-Changes
shell>del Module-Metadata-Changes-1.00.ppd
shell>del PPM-Module-Metadata-Changes-1.00.tar.gz


CHANGES
Revision history for Perl extension Manage::Module::Changes.

2.05  Unreleased
	- Rename test files from \d\d.*.t to *.t.

2.04  Mon Mar 26 11:31:00 2012
	- Stop uninit warning when a line being tested for a datetime contains just 1 field.
	- Output a '-' as documented, rather than a '.', at the start of any line which happens to start with
		Config::IniFiles' reserved token \E\O\T.
	- Add META.json to MANIFEST.

2.03  Wed Feb 23 17:21:00 2011
	- Add missing dependency Hash::FieldHash to Build.PL and Makefile.PL. Re-generate metatdata files.

2.02  Tue Feb 22  9:07:00 2011
	- Add missing dependency Perl6::Slurp to Build.PL and Makefile.PL. Re-generate metatdata files.

2.01  Mon Feb 21 18:30:00 2011
	- Allow for a line in CHANGES to contain just 1 long token, without outputting a warning under recent Perls.

2.00  Tue Feb 15 14:59:00 2011
	- Rewrite to allow client access to the memory-based version of Changelog.ini.
	- Add scripts/report.names.pl to demonstrate this feature.
	- new() now takes a hash, not a hashref, since I've adopted the fabulous Hash::FieldHash.
	- Replace eval{} with Try::Tiny.
	- Only remove commas from lines which match /^\d/, since the digit(s) are assumed to be a version #,
	  and hence the commas are assumed to be embedded in dates.
	- Discontinue convert() and convert_body(), which were available in V 1.
	- What used to be convert() is now writer(reader() ).
	- What used to be convert_body() is now transform().
	- The default input file (under -c) is CHANGES, not ./CHANGES.
	- The default input file (without -c) is Changelog.ini, not ./Changelog.ini.
	- Add warnings to docs that config(), get_latest_release() and get_latest_version() must be called after read().
	- Replace /usr/bin/perl with /usr/bin/env perl.
	- Replace common::sense with use strict and use warnings, to get uninit var warnings.
	- Replace Carp::croak with die.
	- Use real version numbers instead of 0 in the required modules lists in Build.PL and Makefile.PL.

1.08  Sun Feb 21 12:56:13 2010
	- Remove text 'All rights reserved' (for Debian licensing).
	- Remove POD heads 'Required Modules' and 'Changes'.

1.07  Wed Feb 10 13:53:00 2010
	- Add META.yml. Update MANIFEST.SKIP. Add MANIFEST and MYMETA.yml.

1.06  Thu Nov 12 10:19:00 2009
	  - Run dos2unix on all files.

1.05  Thu Jul  8 10:08:00 2009
	  - Rename t/Changes to t/Non.standard.name because under MS Windows there is no difference
	  	between the name t/Changes and the other test file t/CHANGES
	  - Patch t/02.convert.t to match
	  - Use File::HomeDir rather than relying on $ENV{'HOME'}

1.04  Wed Feb 18 11:46:00 2009
	  - Output 'Changelog.*' rather than 'Configlog.*' in the [Module] section of Changelog.ini
	  - Rename Changes.txt to CHANGES because search.cpan.org won't display Changes.txt
	  - Rename t/*_*.t to t/*.*.t (since I prefer '.' to '_' in file names, because '_' is a shift char)

1.03  Tue Jul 22 14:41:00 2008
	  - Rename CHANGES to Changes.txt to match my other modules

1.02  Wed Jul 16 17:14:00 2008
	  - Patch Build.PL and Makefile.PL to depend on DateTime V 0.42 because a user (thanx Mark)
	  	has reported a failed test with an earlier version

1.01  Fri May 16 09:12:00 2008
	  - Add Config::IniFiles to Build.PL and Makefile.PL

1.00  Fri May 02 15:15:45 2008
	  - Original version


Build.PL
use Module::Build;

Module::Build -> new
(
 module_name    => 'Module::Metadata::Changes',
 license        => 'artistic_2',
 dist_abstract  => "Manage a module's machine-readable CHANGES file",
 dist_author    => 'Ron Savage <ron@savage.net.au>',
 build_requires =>
 {
	 Test::More=> 0.96,
#	 Test::Pod  => 1.45, # Make it optional. See t/pod.t
 },
 configure_requires =>
 {
 	 Module::Build => 0.3800,
 },
 requires =>
 {
	 Config::IniFiles            => 2.66,
	 DateTime                    => 0.66,
	 DateTime::Format::HTTP      => 0.40,
	 DateTime::Format::Strptime  => 1.5000,
	 DateTime::Format::W3CDTF    => 0.06,
	 Hash::FieldHash             => 0.10,
	 HTML::Entities::Interpolate => 1.03,
	 HTML::Template              => 2.9,
	 Perl6::Slurp                => 0.03,
	 version                     => 0.82,
 },
) -> create_build_script();

print "\a\n";
print "* Don't forget to install the CSS and *.tmpl files, which are \n";
print "* shipped in the htdocs/ directory within the distro, if you \n";
print "* wish to utilize the HTML output options of ini.report.pl. \n";
print "* Run ini.report.pl -h for details. \n";
print "\n";


Makefile.PL
use ExtUtils::MakeMaker;

# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.

WriteMakefile
(
 ($] ge '5.005') ?
 (
  AUTHOR   => 'Ron Savage (ron@savage.net.au)',
  ABSTRACT => "Manage a module's machine-readable CHANGES file",
 ) : (),
 clean =>
 {
	 FILES => 'blib/* Makefile MANIFEST Module-Metadata-Changes-*'
 },
 dist =>
 {
	 COMPRESS => 'gzip',
	 SUFFIX   => 'gz'
},
 DISTNAME  => 'Module-Metadata-Changes',
 NAME      => 'Module::Metadata::Changes',
	LICENSE   => 'artistic_2',
 PL_FILES  => {},
 PREREQ_PM =>
 {
	 Config::IniFiles            => 2.66,
	 DateTime                    => 0.66,
	 DateTime::Format::HTTP      => 0.40,
	 DateTime::Format::Strptime  => 1.5000,
	 DateTime::Format::W3CDTF    => 0.06,
	 Hash::FieldHash             => 0.10,
	 HTML::Entities::Interpolate => 1.03,
	 HTML::Template              => 2.9,
	 Perl6::Slurp                => 0.03,
	 Test::More                => 0.96,
#	 Test::Pod                   => 1.45, # Make it optional. See t/pod.t
	 version                     => 0.82,
 },
 VERSION_FROM => 'lib/Module/Metadata/Changes.pm',
);

print "\a\n";
print "* Don't forget to install the CSS and *.tmpl files, which are \n";
print "* shipped in the htdocs/ directory within the distro, if you \n";
print "* wish to utilize the HTML output options of ini.report.pl. \n";
print "* Run ini.report.pl -h for details. \n";
print "\n";
